### PR TITLE
Adds guiding principles of repository

### PR DESCRIPTION
**Pull request by stevenleeg**
Hi everyone!

Our team is hoping to start envisioning a fruitful future for this repository in the token ecosystem by defining some guiding principles that differentiate our EIP20 implementation (and any new standard implementations) from others in the space. This PR defines what we think could be a solid set of guidelines which, once accepted by the community and merged, could pave the way for our team to move forward with development and improvements.

Please let us know what you think! We'd love to help breath fresh air into this project and see this as a first step in getting there.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,6 +5,15 @@ This repo contains Solidity smart contract code for simple, standards-compliant
 
 The repo currently implements [EIP20](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-20-token-standard.md) tokens, and more may be added in the future.
 
+## Guiding Principles
+While other projects provide custom and feature-complete token implementations, the contracts found in this repository attempt to suit a slightly different purpose: to serve as the baseline reference implementations of token standards and to capture their essence. We strive to eschew bells and whistles and instead provide minimalist implementations that maximize learning and provide sturdy foundations. Each token implementation shall be accompanied by observations and dialectic regarding its use and evolution across production-grade projects.
+
```
Comment (maurelian) [inline README.md]: What do you mean by 'observations and dialectic'? Would that appear as comments in the code?

Comment (maurelian) [reply README.md]: Is [dialectic](https://en.wikipedia.org/wiki/Dialectic) really the right term for that?

Comment (stevenleeg) [reply README.md]: Happy to change it to be more clear. What if we reworded that sentance to:

> Contracts shall be thoroughly documented, detailing the implementation's rationale and evolution as it has been used across production-grade projects.

**Comment by maurelian** [comment]
That sounds good!

>

**Comment by maurelian** [comment]
I made that change to get it out of my inbox. :)

**Comment by stevenleeg** [comment]
Damn, beat me to it! Glad to see we got this merged and looking forward to the future of the project 😄

**Pull request merged**